flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/SessionReconciler.java

```diff
@@ -157,8 +159,23 @@ public DeleteControl cleanup(FlinkDeployment flinkApp, Context context) {
                             sessionJobs.stream()
                                     .map(job -> job.getMetadata().getName())
                                     .collect(Collectors.toList()));
-            // TODO generate error events for this
-            LOG.warn(error);
+            var events =
+                    EventUtils.getEvents(flinkApp, kubernetesClient).getItems().stream()
+                            .filter(event -> event.getMessage().equals(error))
+                            .collect(Collectors.toList());
+            Event lastEvent = null;
+            if (events.isEmpty()) {
+                LOG.warn(error);
+            } else {
+                lastEvent = events.get(0);
+            }
```

Comment:
Can we move this logic into the `EventUtils`?

Reply:
Here is an example for reference: https://github.com/EnMasseProject/enmasse/blob/master/k8s-api/src/main/java/io/enmasse/k8s/api/KubeEventLogger.java

Reply:
+1 for this suggestion, it makes for a nice self-contained logic that will be very easy to use everywhere.

Reply:
Cool, I will update accordingly